src/main/java/io/cryostat/net/web/http/api/v2/RulesPostHandler.java

```diff
@@ -138,8 +138,15 @@ public boolean isOrdered() {
     @Override
     public IntermediateResponse<String> handle(RequestParameters params) throws ApiException {
         Rule rule;
-        HttpMimeType mime =
-                HttpMimeType.fromString(params.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+        String contentType = params.getHeaders().get(HttpHeaders.CONTENT_TYPE);
+
+        if (contentType == null) {
+            contentType = "";
+        }
```

Comment:
I think we can collapse the conditions here:

```java
if (contentType != null && contentType.contains(";"))
```

This way the `null` case is handled by the `HttpMimeType.fromString()` later on, and the non-null case is handled here.

Alternatively, you can guard against the null case like this:

```java
String contentType = StringUtils.defaultString(params.getHeaders().get(HttpHeaders.CONTENT_TYPE));
```

This way the `contentType` will never be `null`.

Reply:
ok